src/source/PeerConnection/Rtcp.c

```diff
@@ -2,31 +2,125 @@
+}
+
 STATUS onRtcpPacket(PKvsPeerConnection pKvsPeerConnection, PBYTE pBuff, UINT32 buffLen)
 {
     STATUS retStatus = STATUS_SUCCESS;
+    PKvsRtpTransceiver pTransceiver = NULL;
     RtcpPacket rtcpPacket;
-    UINT32 currentOffset = 0;
+    UINT8 fractionLost;
+    UINT32 senderSSRC, rtpTs, packetCnt, octetCnt, ssrc1, cumulativeLost, extHiSeqNumReceived, interarrivalJitter, lastSR, delaySinceLastSR;
+    UINT32 rttPropDelay, rttPropDelayMsec;
+    UINT64 ntpTime;
+    UINT64 currentTimeNTP = convertTimestampToNTP(GETTIME());
+    UINT32 currentOffset  = 0;
 
     CHK(pKvsPeerConnection != NULL && pBuff != NULL, STATUS_NULL_ARG);
 
     while (currentOffset < buffLen) {
         CHK_STATUS(setRtcpPacketFromBytes(pBuff + currentOffset, buffLen - currentOffset, &rtcpPacket));
 
-        if (rtcpPacket.header.packetType == RTCP_PACKET_TYPE_GENERIC_RTP_FEEDBACK && rtcpPacket.header.receptionReportCount == RTCP_FEEDBACK_MESSAGE_TYPE_NACK) {
-            CHK_STATUS(resendPacketOnNack(&rtcpPacket, pKvsPeerConnection));
-        } else if (rtcpPacket.header.packetType == RTCP_PACKET_TYPE_PAYLOAD_SPECIFIC_FEEDBACK &&
-                   rtcpPacket.header.receptionReportCount == RTCP_FEEDBACK_MESSAGE_TYPE_APPLICATION_LAYER_FEEDBACK &&
-                   isRembPacket(rtcpPacket.payload, rtcpPacket.payloadLength) == STATUS_SUCCESS)
-        {
-            CHK_STATUS(onRtcpRembPacket(&rtcpPacket, pKvsPeerConnection));
-        } else if (rtcpPacket.header.packetType == RTCP_PACKET_TYPE_PAYLOAD_SPECIFIC_FEEDBACK &&
-                   rtcpPacket.header.receptionReportCount == RTCP_PSFB_PLI)
-        {
-            CHK_STATUS(onRtcpPLIPacket(&rtcpPacket, pKvsPeerConnection));
+        switch (rtcpPacket.header.packetType) {
+            case RTCP_PACKET_TYPE_GENERIC_RTP_FEEDBACK:
+                if (rtcpPacket.header.receptionReportCount == RTCP_FEEDBACK_MESSAGE_TYPE_NACK)
+                    CHK_STATUS(resendPacketOnNack(&rtcpPacket, pKvsPeerConnection));
+                break;
+            case RTCP_PACKET_TYPE_PAYLOAD_SPECIFIC_FEEDBACK:
+                if (rtcpPacket.header.receptionReportCount == RTCP_FEEDBACK_MESSAGE_TYPE_APPLICATION_LAYER_FEEDBACK &&
+                    isRembPacket(rtcpPacket.payload, rtcpPacket.payloadLength) == STATUS_SUCCESS) {
+                    CHK_STATUS(onRtcpRembPacket(&rtcpPacket, pKvsPeerConnection));
+                } else if (rtcpPacket.header.receptionReportCount == RTCP_PSFB_PLI) {
+                    CHK_STATUS(onRtcpPLIPacket(&rtcpPacket, pKvsPeerConnection));
+                } else {
+                    DLOGW("unhandled packet type RTCP_PACKET_TYPE_PAYLOAD_SPECIFIC_FEEDBACK");
+                }
+                break;
+            case RTCP_PACKET_TYPE_SENDER_REPORT:
+                // https://tools.ietf.org/html/rfc3550#section-6.4.1
+                if (rtcpPacket.payloadLength == RTCP_PACKET_SENDER_REPORT_MINLEN) {
+                    senderSSRC = getUnalignedInt32BigEndian(rtcpPacket.payload);
```

Comment:
same comment about the arranging the bits

Reply:
i am pretty sure i'll be changing this part of code quite a bit soon